File: TwoSumOnePassHash/twosum.py
def twoSum(nums, target):
    # Declare ang map
    num_map = {}  
    # Iterate the list
    for i, num in enumerate(nums):
        
        complement = target - num  
        
        if complement in num_map:  
            return [num_map[complement], i]
        
        num_map[num] = i  

    return []  

File: TwoSumOnePassHash/twosum.php
function twoSum($nums, $target) {
    $map = [];

    $n = count($nums);

    for($i = 0; $i < $n; $i++) {
        $complement = $target - $nums[$i];  
        
        if(array_key_exists($complement, $map)) {
            return [$map[$complement], $i];  
        }
        
        $map[$nums[$i]] = $i;  
    }
    
    return [];  
}
